### PR TITLE
Activate: Native store creation flow improvements

### PR DESCRIPTION
**Pull request by 0nko**
This PR started mainly as a way to dynamically load eCommerce plan data from the API but it also adds a couple of more improvements. 

- The plan price, name and period is loaded dynamically from the `v1.5/plans` endpoint
- The flow can now recover from a `site already exists` error if the site belongs to the current user (see videos below)
- To align with iOS, there is a new `web_checkout` step in tracking
- To be able implement the point above, the cart checkout logic was moved under the Plan purchase screen, which actually makes more sense
- Improved error screen UI
- Improved the scaling of the new store preview (more zoomed out, without distracting dialogs)
- Refactoring of common code to improve reusability (e.g. error and progress screens)
- `NewStore` data is cleared after the store creation is completed, so that it's ready for a new store creation
- The last navigation step to the Installation screen clears the back stack, so that when a new site is created and a user taps the back button, they're redirected back to the site picker, not the previous store creation step
- A couple of minor UI tweaks

**To test:**
1. Start the store creation flow
2. Keep tapping on Continue until the webview checkout step
3. Tap the back button
4. Then tap on Continue to start the checkout page again
5. Verify that the checkout page is displayed again
6. Complete the purchase flow
7. Verify the following track events are logged:
```
🔵 Tracked: site_creation_step, Properties: {"step":"store_name","is_debug":true}
🔵 Tracked: site_creation_step, Properties: {"step":"domain_picker","is_debug":true}
🔵 Tracked: site_creation_step, Properties: {"step":"store_summary","is_debug":true}
🔵 Tracked: site_creation_step, Properties: {"step":"plan_purchase","is_debug":true}
🔵 Tracked: site_creation_step, Properties: {"step":"web_checkout","is_debug":true}
🔵 Tracked: site_creation_step, Properties: {"step":"store_installation","is_debug":true}
🔵 Tracked: login_woocommerce_site_created, Properties: {"source":"switching_store","url":"{{site_url}}","flow":"native","blog_id":{{site_id}},"is_wpcom_store":false,"is_debug":true}
```

https://user-images.githubusercontent.com/1522856/205019856-b71307f2-32a7-4fdd-b93d-998c96307d59.mp4

https://user-images.githubusercontent.com/1522856/205019932-655d63a4-d175-4ac5-9962-42c14b0a2178.mp4

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/StoreCreationErrorScreen.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.woocommerce.android.R.dimen
+import com.woocommerce.android.R.drawable
+import com.woocommerce.android.R.string
+import com.woocommerce.android.ui.compose.component.WCColoredButton
+
+@Composable
+fun StoreCreationError(
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/StoreCreationErrorScreen.kt]: Np, might make sense to treat this as a screen and name the root composable with the "screen" suffix to match the filename

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/StoreCreationErrorScreen.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                        .fillMaxWidth()
+                        .padding(
+                            vertical = dimensionResource(id = dimen.major_150),
+                            horizontal = dimensionResource(id = dimen.major_300)
+                        )
+                )
+            }
+        }
+    }
+}
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/StoreCreationErrorScreen.kt:103]:
``` suggestion

@Preview(name = "dark", uiMode = Configuration.UI_MODE_NIGHT_YES)
@Preview(name = "light", uiMode = Configuration.UI_MODE_NIGHT_NO)
@Preview(name = "small screen", device = Devices.PIXEL)
@Preview(name = "mid screen", device = Devices.PIXEL_4)
@Preview(name = "large screen", device = Devices.NEXUS_10)
@Composable
fun StoreCreationErrorPreview() {
    WooThemeWithBackground {
        StoreCreationError(
            errorType = StoreCreationErrorType.SITE_CREATION_FAILED,
            onArrowBackPressed = {},
            message = "Error creating store",
            onRetryButtonClicked = {}
        )
    }
}

```

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/StoreCreationProgressScreen.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        if (message != null) {
+            Text(
+                text = message,
+                style = MaterialTheme.typography.h6,
+                modifier = Modifier.padding(16.dp)
+            )
+        }
+    }
+}
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/StoreCreationProgressScreen.kt]: If this is generic enough might make sense to move it to `ui/compose/component` ?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/StoreCreationResult.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.woocommerce.android.ui.login.storecreation
+
+sealed interface StoreCreationResult<T> {
+    data class Success<T>(val data: T) : StoreCreationResult<T>
+    data class Failure<T>(val type: StoreCreationErrorType, val message: String? = null) : StoreCreationResult<T>
+}
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/StoreCreationResult.kt:6]: Nice!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/installation/InstallationScreen.kt [hunk 5]
```diff
@@ -32,76 +32,40 @@ import androidx.compose.ui.res.colorResource
...
-            } else if (url == viewState.exitTriggerKeyword) {
-                onExitTriggered()
-            }
-        },
-        modifier = Modifier.fillMaxSize()
-    )
-}
-
 @Composable
 private fun InstallationSummary(url: String, viewModel: InstallationViewModel) {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/installation/InstallationScreen.kt]: Np, I'd avoid passing the viewModel here and instead pass only the function references you need `viewModel::onManageStoreButtonClicked` and `viewModel::onShowPreviewButtonClicked`

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Awesome job @0nko 🙌🏼 

Code looks great and works like a charm. I left a couple of nitpicks. Nothing blocker.

**Comment by 0nko** [comment]
Thanks for the code review, @JorgeMucientes. I've made the suggested changes.

**Pull request merged**